### PR TITLE
refactor: remove various go template references in cse_install

### PR DESCRIPTION
**Pull request by alexeldeib**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: parts/linux/cloud-init/artifacts/cse_install.sh [hunk 1]
```diff
@@ -349,25 +346,25 @@ cleanUpImages() {
...
+        if [ "${NEEDS_CONTAINERD}" == "true" ]; then
+            if [[ "${CLI_TOOL}" == "crictl" ]]; then
+                images_to_delete=$(crictl images | awk '{print $1":"$2}' | grep -vE "${KUBERNETES_VERSION}$|${KUBERNETES_VERSION}.[0-9]+$|${KUBERNETES_VERSION}-|${KUBERNETES_VERSION}_" | grep ${targetImage} | tr ' ' '\n')
+            else
+                images_to_delete=$(ctr --namespace k8s.io images list | awk '{print $1}' | grep -vE "${KUBERNETES_VERSION}$|${KUBERNETES_VERSION}.[0-9]+$|${KUBERNETES_VERSION}-|${KUBERNETES_VERSION}_" | grep ${targetImage} | tr ' ' '\n')
+            fi
         else
-            images_to_delete=$(ctr --namespace k8s.io images list | awk '{print $1}' | grep -vE "${KUBERNETES_VERSION}$|${KUBERNETES_VERSION}.[0-9]+$|${KUBERNETES_VERSION}-|${KUBERNETES_VERSION}_" | grep ${targetImage} | tr ' ' '\n')
+            images_to_delete=$(docker images --format '{{OpenBraces}}.Repository{{CloseBraces}}:{{OpenBraces}}.Tag{{CloseBraces}}' | grep -vE "${KUBERNETES_VERSION}$|${KUBERNETES_VERSION}.[0-9]+$|${KUBERNETES_VERSION}-|${KUBERNETES_VERSION}_" | grep ${targetImage} | tr ' ' '\n')
```
Comment (alexeldeib) [inline parts/linux/cloud-init/artifacts/cse_install.sh:356]: idk why github is highlighting this diff, this is not the diff...

Comment (UtheMan) [reply parts/linux/cloud-init/artifacts/cse_install.sh:356]: looks pretty different to me... 😄

Comment (alexeldeib) [reply parts/linux/cloud-init/artifacts/cse_install.sh:356]: lol touche, but this specific line is the same, just bad word splitting in the diff algo

File: parts/linux/cloud-init/artifacts/cse_install.sh [hunk 2]
```diff
@@ -89,7 +89,6 @@ downloadAzureCNI() {
     CNI_TGZ_TMP=${VNET_CNI_PLUGINS_URL##*/} # Use bash builtin ## to remove all chars ("*") up to the final "/"
     retrycmd_get_tarball 120 5 "$CNI_DOWNLOADS_DIR/${CNI_TGZ_TMP}" ${VNET_CNI_PLUGINS_URL} || exit $ERR_CNI_DOWNLOAD_TIMEOUT
 }
-{{- if NeedsContainerd}}
```
Comment (UtheMan) [inline parts/linux/cloud-init/artifacts/cse_install.sh:92]: why were this function in a template block before (and those below too i guess)?

Comment (alexeldeib) [reply parts/linux/cloud-init/artifacts/cse_install.sh:92]: reduced size of cloud init payload

we used to run into a lot of issues with that in aks-engine days

ironically, me making all these changes to move them into vhd will eliminate that forever, but aks engine supported vanilla ubuntu an other stuff too where that wasn't an option

**Pull request merged**